### DISCUSSION TITLE
Integrations with Application Performance Monitoring

### DISCUSSION THREAD

**Discussion by PranavSathy** [prisma/prisma#4319] [Q&A] [answered]
Hi Everyone,

Been a huge fan of Prisma from the early days (GraphCool)! I was curious however, I wanted to understand if a NewRelic or DataDog integration on our NodeJS Express server while utilizing Prisma would still show performance characteristics of the underlying SQL queries?

https://docs.newrelic.com/docs/agents/nodejs-agent/getting-started/compatibility-requirements-nodejs-agent

The document above seems to suggest that as long as `prisma` uses the `pg` driver, that this should be the case. 

Thanks in advance!

**Comment by ryands17** [comment, marked as answer]
@PranavSathy Prisma doesn't use the Pg driver as Prisma has its own query engine under the hood. In this particular case, you would need to capture queries yourself. For e.g. you can do something like this:

```ts
const prisma = new PrismaClient({
  log: [
    'query',
    {
      emit: 'event',
      level: 'query',
    },
  ],
})

prisma.$on("query", async (e) => {
  // call the Newrelic/DataDog API here
  console.log(`${e.query} ${e.params}`)
});
```

Hope this helps :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
